### ISSUE TITLE
Comfort function for setting the mode registers 

### ISSUE THREAD

**Issue by Luxovious** [RobTillaart/PCA9634#7]
Hi Rob,
as I'm very happy, using your library for a while now I found a comfort issue that keeps bugging me a little. Most of the times, when I use the PCA9634 chip, I use it with external drivers, as I want to drive led strips or similar with it. Furthermore, I sometimes like to use the subadresses on a few chips. Both things can be configured in the MODE1 and MODE 2 registers of the chip. I'm thankful, that you implemented the PCA9634::writeMode - function, as I can use it for that purpose.

In my recent project I configured the chip as follows, since it was the "fastest" way for me.
ledArray.writeMode(PCA9634_MODE2,0b010101);

However, I often have to take a look in the datasheet to set the correct bits high/low for the configuration to work. It would be very comfortable, if there was a simpler way with "words" rather than bit manipulation. Others might find it helpful to have something like:
ledArray.configure(...);
combined with "words" such as STOP, TOTEM_POLE, INVERTED, etc. which resolve to give the correct configuration.

A comment I thought was usefull for your library (line 112, above the "writeMode" function):
// use to write to the two mode registers. Needed for configuring the output (check datasheet)

**Comment by RobTillaart** [comment]
Hi, I got serious ill this weekend and it might take weeks before being on my feet again.

The issue is valid IMHO so it is added to my list.
Please be patient..

**Comment by Luxovious** [comment]
Get well soon!

BTW: I'm going to check out, if the library works well with subadresses. In case I run into issues, I'll report.

**Comment by RobTillaart** [comment]
@Luxovious 
would it help to define the following constants, that can be OR-ed to get the mask for the MODEx register?

```cpp
// Configuration bits MODE1 register
#define PCA9634_MODE1_AUTOINCR2     0x80  // RO, 0 = disable  1 = enable
#define PCA9634_MODE1_AUTOINCR1     0x40  // RO, bit1
#define PCA9634_MODE1_AUTOINCR0     0x20  // RO  bit0
#define PCA9634_MODE1_SLEEP         0x10  // 0 = normal       1 = sleep
#define PCA9634_MODE1_SUB1          0x08  // 0 = disable      1 = enable
#define PCA9634_MODE1_SUB2          0x04  // 0 = disable      1 = enable
#define PCA9634_MODE1_SUB3          0x02  // 0 = disable      1 = enable
#define PCA9634_MODE1_ALLCALL       0x01  // 0 = disable      1 = enable

// Configuration bits MODE2 register
#define PCA9634_MODE2_BLINK         0x20  // 0 = dim          1 = blink
#define PCA9634_MODE2_INVERT        0x10  // 0 = normal       1 = inverted
#define PCA9634_MODE2_STOP          0x08  // 0 = on STOP      1 = on ACK
#define PCA9634_MODE2_TOTEMPOLE     0x04  // 0 = open drain   1 = totem-pole
```

PCA9635 should get similar constants 

```cpp
ledArray.writeMode(PCA9634_MODE2, 0b00110100);

// would become

uint8_t mode2mask = PCA9634_MODE2_BLINK | PCA9634_MODE2_INVERT | PCA9634_MODE2_TOTEMPOLE;
ledArray.writeMode(PCA9634_MODE2, mode2mask);
```

optional wrapper functions can be made

```cpp
//  convenience wrappers
uint8_t  setMode1(uint8_t value) { return writeMode(PCA9634_MODE1, value); };
uint8_t  setMode2(uint8_t value) { return writeMode(PCA9634_MODE2, value); };
uint8_t  getMode1()              { return readMode(PCA9634_MODE1); };
uint8_t  getMode2()              { return readMode(PCA9634_MODE2); };
```
making a function that handles both registers as one is possible but I do not like that as it "drifts away" from the datasheet.

**Comment by RobTillaart** [comment]
@Luxovious 
Created a branch for the above - https://github.com/RobTillaart/PCA9634/tree/develop
(enough for today)

**Comment by RobTillaart** [comment]
@Luxovious 
Any comments on the  solution?

**Comment by Luxovious** [comment]
Hi Rob,

I completely missed your post/branch. Thank you for the reminder. Since I just came back from a trip, I'll check within the next few days during the week! As soon as I find bugs/etc. or if everything works perfect I'll get back at you.

Have a nice Sunday evening!
Johannes

**Comment by RobTillaart** [comment]
@Luxovious 
just another reminder

**Comment by Luxovious** [comment]
Hi Rob,
thank you for your reminder. Today I finally found some time to play with your work. Unfortunately, I found your comment with "mode2mask" just now. I spend too much time trying to figure out how to put something together with the same result. This should definately be part of your examples as this is a perfect solution. The convienence wrappers are nice, however your new #defines for the register descriptions are already solving this request! Thank you a lot.

I found other little things, but I will document them seperately, to keep things clean :-)

**Comment by RobTillaart** [comment]
Added the **mode2_mask** snippet above in the readme.md file

**Comment by RobTillaart** [comment]
Will merge later today,

**Comment by RobTillaart** [comment]
solved in 0.1.2
